File: ruby/arrays/prepending_elements.rb
#!/usr/bin/ruby2.7

array = [2, 4, 6]

puts("Initial state:\n\n")
puts(array.inspect, "\n")

# prepend

array.prepend(0)

puts("After prepend by array.prepend(0):\n\n")
puts(array.inspect, "\n")

array.prepend(-6, -4, -2)

puts("After prepend by array.prepend(-6, -4, -2):\n\n")
puts(array.inspect, "\n")

# unshift

array.unshift(-8)

puts("After prepend by array.unshift(-8):\n\n")
puts(array.inspect, "\n")

array.unshift(-14, -12, -10)

puts("After prepend by array.unshift(-14, -12, -10):\n\n")
puts(array.inspect, "\n")

File: javascript/arrays/operations_on_elements/prepending_elements.js
#!/usr/bin/env node

let array = [2, 4, 6];

console.log("Initial state:\n");
console.log(array); console.log();

let r = array.unshift(0);

console.log("After prepend by array.unshift(0):\n");
console.log(array); console.log();
console.log(`Returned value: ${r}\n`);

r = array.unshift(-6, -4, -2);

console.log("After prepend by array.unshift(-6, -4, -2):\n");
console.log(array); console.log();
console.log(`Returned value: ${r}\n`);
